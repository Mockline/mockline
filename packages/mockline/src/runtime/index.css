@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

