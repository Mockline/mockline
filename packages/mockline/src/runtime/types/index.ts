import type { Color } from '../utils/colors'

export * from '../components/elements/Button.vue'

export type ModuleOptions = {
  /**
   * Prefix for all components
   */
  prefix?: string,
  /**
   * Enable Nuxt Content
   * @default false
   */
  content?: boolean
  /**
   * The colors to use
   */
  colors?: Color[]
  /**
   * Enable Color Transitions
   * @default false
   */
  transitions?: boolean
}
