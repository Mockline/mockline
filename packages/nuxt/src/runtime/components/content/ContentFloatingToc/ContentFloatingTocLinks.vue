<script setup lang="ts">
import type { TocLink } from '@nuxt/content'
import { useRouter } from 'vue-router'

const { links = [], isHover, isMobile } = defineProps<{
  links: TocLink[]
  active?: string
  isHover: boolean
  activeHeadings: string[]
  isMobile: boolean
}>()

const router = useRouter()

const scrollToHeading = (id: string): void => {
  if (!isMobile || (isMobile && isHover)) {
    const encodedId = encodeURIComponent(id)

    const targetElement = document.getElementById(id)
    if (targetElement) {
      const contentPanel = document.getElementById('app-main-content')
      if (contentPanel) {
        const targetPosition = targetElement.offsetTop

        contentPanel.scrollTo({
          top: targetPosition - 20,
          behavior: 'smooth'
        })
      }
    }
    router.push(`#${encodedId}`)
  }
}
</script>

<template>
  <div>
    <Transition
      enter-active-class="transition-all duration-100 ease-in-out"
      enter-from-class="opacity-0"
      enter-to-class="opacity-100"
      leave-active-class="transition-all duration-100 ease-in-out"
      leave-from-class="opacity-100"
      leave-to-class="opacity-0"
      mode="out-in"
    >
      <ul v-if="links?.length && isHover" class="space-y-2">
        <li
          v-for="link in links"
          :key="link.text"
          :class="[
            link.depth === 3 ? 'ml-4' : ''
          ]"
        >
          <a
            class="block truncate text-xs/4 transition-all duration-500 ease-in-out"
            :class="activeHeadings.includes(link.id) ? 'text-accent' : 'text-[var(--color-text-tertiary)] hover:text-[var(--color-text-primary)]'"
            :href="`#${link.id}`"
            @click.prevent="scrollToHeading(link.id)"
          >
            {{ link.text }}
          </a>

          <MContentFloatingTocLinks
            v-if="link.children"
            :links="link.children"
            :is-hover
            :active-headings
            :is-mobile
          />
        </li>
      </ul>
      <div v-else-if="links?.length" class="space-y-4">
        <div
          v-for="link in links"
          :key="link.text"
          class="flex flex-col items-end gap-1"
        >
          <div
            class="h-[4px] transition-all duration-500 ease-in-out"
            :class="[
              activeHeadings.includes(link.id)
                ? 'w-11 bg-accent'
                : 'bg-[var(--color-bg-quaternary)]',
              link.depth === 1 ? 'w-12' : link.depth === 2 ? 'w-10' : 'w-8',
              isMobile ? 'cursor-default' : 'cursor-pointer'
            ]"
          />

          <MContentFloatingTocLinks
            v-if="link.children"
            :links="link.children"
            :is-hover
            :active-headings
            :is-mobile
          />
        </div>
      </div>
    </Transition>
  </div>
</template>
