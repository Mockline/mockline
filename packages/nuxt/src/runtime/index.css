@import "@mockline/themes";
@import "./mockline.css";
