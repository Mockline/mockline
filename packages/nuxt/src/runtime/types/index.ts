// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
