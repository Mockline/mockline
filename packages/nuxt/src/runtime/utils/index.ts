export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
