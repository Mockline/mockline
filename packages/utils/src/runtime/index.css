@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
