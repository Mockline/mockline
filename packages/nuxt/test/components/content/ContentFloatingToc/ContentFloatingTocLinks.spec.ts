import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import ContentFloatingTocLinks from '../../src/runtime/components/content/ContentFloatingToc/ContentFloatingTocLinks.vue'

describe('ContentFloatingTocLinks', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(ContentFloatingTocLinks)
    expect(html.html()).toMatchSnapshot()
  })
})
