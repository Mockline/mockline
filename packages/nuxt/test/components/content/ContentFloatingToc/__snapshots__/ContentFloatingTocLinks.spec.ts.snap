// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentFloatingTocLinks > can mount component 1`] = `
"<div id="test-wrapper">
  <div>
    <transition-stub enteractiveclass="transition-all duration-100 ease-in-out" enterfromclass="opacity-0" entertoclass="opacity-100" leaveactiveclass="transition-all duration-100 ease-in-out" leavefromclass="opacity-100" leavetoclass="opacity-0" mode="out-in" appear="false" persisted="false" css="true">
      <!--v-if-->
    </transition-stub>
  </div>
</div>"
`;

exports[`ContentFloatingTocLinks > renders <MContentFloatingTocLinks :links="[{id: 'id', text: 'text', depth: 1}]" :activeHeadings="['id']" /> correctly 1`] = `
"<div>
  <transition-stub enteractiveclass="transition-all duration-100 ease-in-out" enterfromclass="opacity-0" entertoclass="opacity-100" leaveactiveclass="transition-all duration-100 ease-in-out" leavefromclass="opacity-100" leavetoclass="opacity-0" mode="out-in" appear="false" persisted="false" css="true">
    <div class="space-y-4">
      <div class="flex flex-col items-end gap-1">
        <div class="h-[4px] transition-all duration-500 ease-in-out w-11 bg-accent w-12 cursor-pointer"></div>
        <!--v-if-->
      </div>
    </div>
  </transition-stub>
</div>"
`;

exports[`ContentFloatingTocLinks > renders basic case correctly 1`] = `
"<div>
  <transition-stub enteractiveclass="transition-all duration-100 ease-in-out" enterfromclass="opacity-0" entertoclass="opacity-100" leaveactiveclass="transition-all duration-100 ease-in-out" leavefromclass="opacity-100" leavetoclass="opacity-0" mode="out-in" appear="false" persisted="false" css="true">
    <div class="space-y-4">
      <div class="flex flex-col items-end gap-1">
        <div class="h-[4px] transition-all duration-500 ease-in-out w-11 bg-accent w-12 cursor-pointer"></div>
        <!--v-if-->
      </div>
    </div>
  </transition-stub>
</div>"
`;
